newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/ExtensionDependenciesEditor.js

```diff
@@ -0,0 +1,168 @@
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+
+import Window from '../../Utils/Window';
+import { mapVector } from '../../Utils/MapFor';
+import newNameGenerator from '../../Utils/NewNameGenerator';
+import useForceUpdate from '../../Utils/UseForceUpdate';
+import BackgroundText from '../../UI/BackgroundText';
+
+const checkNameExists = (
+  name: string,
+  deps: gdVectorDependencyMetadata
+): boolean => {
+  for (let i = 0; i < deps.size(); i++)
+    if (deps.at(i).getName() === name) return true;
+  return false;
+};
+
+type Props = {| eventsFunctionsExtension: gdEventsFunctionsExtension |};
+
+export const ExtensionDependenciesEditor = ({
+  eventsFunctionsExtension,
+}: Props) => {
+  const deps = eventsFunctionsExtension.getAllDependencies();
+  const forceUpdate = useForceUpdate();
+
+  const addDependency = () => {
+    eventsFunctionsExtension
+      .addDependency()
+      .setName(
+        newNameGenerator('New Dependency', newName =>
+          checkNameExists(newName, deps)
+        )
+      )
+      .setExportName('my-dependency')
+      .setVersion('1.0.0')
+      .setDependencyType('cordova');
+    forceUpdate();
+  };
+
+  return (
+    <>
+      <TableContainer
+        component={({ children }) => <Paper elevation={4}>{children}</Paper>}
+      >
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>
+                <Trans>Name</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Export name</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Version</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Dependency type</Trans>
+              </TableCell>
+              <TableCell>
+                <Trans>Action</Trans>
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {mapVector<gdDependencyMetadata, TableRow>(
+              eventsFunctionsExtension.getAllDependencies(),
+              (dependency, index) => (
+                <TableRow key={dependency.getName()}>
+                  <TableCell>
+                    <SemiControlledTextField
+                      commitOnBlur
+                      value={dependency.getName()}
+                      onChange={newName => {
+                        if (checkNameExists(newName, deps))
+                          Window.showMessageBox(
+                            `This name is already in use! Please use a unique name.`,
+                            'error'
+                          );
+                        else
+                          eventsFunctionsExtension
+                            .getAllDependencies()
+                            .at(index)
+                            .setName(newName) && forceUpdate();
```

Comment:
I highly discourage abusing && like this: 
- If the function returns void/undefined/falsy value, the second instruction won't work
- if someones wants to add an instruction, it's a pain.
- you have a chance to forget the && one day and end up with something executed outside of the if else.

Prefer:
- using a comma if you really really want not to write brackets:
```
if(false) 
  1, 2 
else 
  3, 4
```
- or (better), just use brackets.

In some codebases, not writing brackets is not even allowed, because of the risk of forgetting them when adding an instruction.
I'm not against them, but only if it's for a single line instruction. If it's multiple lines, and there is a else => brackets.